---
title: "SampleCodeForNeoantimon"
author: "T. Hasegawa"
date: "2019/12/13"
output:
  pdf_document: default
  html_document: default
---

<!--
%\VignetteEngine{knitr::knitr}
%\VignetteIndexEntry{Sample Code to Use Neoantimon}
-->

## Data Preparation and Sample Codes for Analysis
```{r Preparation}
#install.packages('devtools');
#library(devtools);
#install_github('hase62/Neoantimon');
#library(Neoantimon);
for(rc in list.files("/Users/takaorihasegawa/Git/Neoantimon/R")) source(paste("/Users/takaorihasegawa/Git/Neoantimon/R", rc, sep = "/"))

```

```{r Sample VCF file}
data("sample_vcf")
head(sample_vcf, row.names = FALSE)
```

```{r Sample VCF (SV Fusion BND) file}
data("sample_sv_bnd")
head(sample_sv_bnd, row.names = FALSE)
```

```{r HLA Table for Class 1}
data("sample_hla_table_c1")
head(sample_hla_table_c1, row.names = FALSE)
```

```{r HLA Table for Class2}
data("sample_hla_table_c2")
head(sample_hla_table_c2, row.names = FALSE)
```

```{r RNA Expression}
data("sample_rna_exp")
head(sample_rna_exp, row.names = FALSE)
```

```{r CopyNumber Information}
data("sample_copynum")
head(sample_copynum, row.names = FALSE)
```

```{r Sample.Result.SNV.HLACLASS1}
data("sample_result_SNV_CLASS1_ALL")
head(sample_result_SNV_CLASS1_ALL, row.names = FALSE)
```

```{r Sample.Result.SNV.HLACLASS2}
data("sample_result_SNV_CLASS2_ALL")
head(sample_result_SNV_CLASS2_ALL, row.names = FALSE)
```

```{r Sample.Result.INDEL.HLACLASS1}
data("sample_result_INDEL_CLASS1_ALL")
head(sample_result_INDEL_CLASS1_ALL, row.names = FALSE)
```

```{r Sample.Result.INDEL.HLACLASS2}
data("sample_result_INDEL_CLASS2_ALL")
head(sample_result_INDEL_CLASS2_ALL)
```

```{r Sample.Result.SV.HLACLASS1}
data("sample_result_SVFusion_CLASS1_ALL")
head(sample_result_SVFusion_CLASS1_ALL, row.names = FALSE)
```

```{r Sample.Result.SV.HLACLASS2}
data("sample_result_SVFusion_CLASS2_ALL")
head(sample_result_SVFusion_CLASS2_ALL)
```

```{r Sample.Result.SeqFragment.HLACLASS1}
data("sample_result_SeqFragment_CLASS1_ALL")
head(sample_result_SeqFragment_CLASS1_ALL, row.names = FALSE)
```

```{r Sample.Result.SeqFragment.HLACLASS2}
data("sample_result_SeqFragment_CLASS2_ALL")
head(sample_result_SeqFragment_CLASS2_ALL)
```

```{r Get Sample and Test Analysis}
TestAnalysis
```

```{r Get SNV Sample 1 in Test Analysis}
  Result_HLA1_SNV <- MainSNVClass1(input_file = "data/sample_vcf.txt",
                                   file_name_in_hla_table = "sample",
                                   hla_file = "data/sample_hla_table_c1.txt",
                                   refflat_file  = "lib/refFlat.txt",
                                   refmrna_file = "lib/refMrna.fa",
                                   rnaexp_file = "data/sample_rna_exp.txt",
                                   netMHCpan_dir = "lib/netMHCpan-4.0/netMHCpan",
                                   depth_tumor_column = 12,
                                   depth_normal_column = 14,
                                   SNPs = "lib/sample.snps.vcf",
                                   multiple_variants = TRUE,
                                   apply_annotation = FALSE,
                                   MHCflurry = "mhctools")
  print(head(Result_HLA1_SNV[[1]]))
```

```{r Get SNV Summary 1 in Test Analysis}
  print(Export_Summary_SNV(Input = Result_HLA1_SNV[[1]], Mut_IC50_th = 500, Wt_IC50_th = 500))
```

```{r Get SNV Sample 2 in Test Analysis}
Result_HLA2_SNV <- MainSNVClass2(input_file = "data/sample_vcf.txt",
                                   file_name_in_hla_table = "sample",
                                   hla_file = "data/sample_hla_table_c2.txt",
                                   refflat_file  = "lib/refFlat.txt",
                                   refmrna_file = "lib/refMrna.fa",
                                   rnaexp_file = "data/sample_rna_exp.txt",
                                   netMHCIIpan_dir = "lib/netMHCIIpan-3.2/netMHCIIpan",
                                   depth_tumor_column = 12,
                                   depth_normal_column = 14,
                                   SNPs = "lib/sample.snps.vcf",
                                   multiple_variants = TRUE,
                                   apply_annotation = FALSE)
  print(head(Result_HLA2_SNV))
```

```{r Get SNV Summary 2 in Test Analysis}
  print(Export_Summary_SNV(Input = Result_HLA2_SNV, Mut_IC50_th = 500, Wt_IC50_th = 500))
```

```{r Get INDEL Sample 1 in Test Analysis}
  Result_HLA1_INDEL <- MainINDELClass1(input_file = "data/sample_vcf.txt",
                                       file_name_in_hla_table = "sample",
                                       hla_file = "data/sample_hla_table_c1.txt",
                                       refflat_file  = "lib/refFlat.txt",
                                       refmrna_file = "lib/refMrna.fa",
                                       rnaexp_file = "data/sample_rna_exp.txt",
                                       netMHCpan_dir = "lib/netMHCpan-4.0/netMHCpan",
                                       depth_tumor_column = 12,
                                       depth_normal_column = 14,
                                       SNPs = "lib/sample.snps.vcf",
                                       multiple_variants = TRUE,
                                       apply_annotation = FALSE,
                                       MHCflurry = "mhctools")
  print(head(Result_HLA1_INDEL[[1]]))
```

```{r Get INDEL Summary 1-1 in Test Analysis}
  print(Export_Summary_IndelSV(Input = Result_HLA1_INDEL[[1]], Mut_IC50_th = 500))
```

```{r Get INDEL Summary 1-2 in Test Analysis}
  print(Export_Summary_IndelSV_perFragments(Input = Result_HLA1_INDEL[[1]], Mut_IC50_th = 500))
```

```{r Get INDEL Sample 2 in Test Analysis}
  Result_HLA2_INDEL <- MainINDELClass2(input_file = "data/sample_vcf.txt",
                                       file_name_in_hla_table = "sample",
                                       hla_file = "data/sample_hla_table_c2.txt",
                                       refflat_file  = "lib/refFlat.txt",
                                       refmrna_file = "lib/refMrna.fa",
                                       rnaexp_file = "data/sample_rna_exp.txt",
                                       netMHCIIpan_dir = "lib/netMHCIIpan-3.2/netMHCIIpan",
                                       depth_tumor_column = 12,
                                       depth_normal_column = 14,
                                       SNPs = "lib/sample.snps.vcf",
                                       multiple_variants = TRUE,
                                       apply_annotation = FALSE)
  print(head(Result_HLA2_INDEL))
```

```{r Get INDEL Summary 2-1 in Test Analysis}
  print(Export_Summary_IndelSV(Input = Result_HLA2_INDEL, Mut_IC50_th = 500))
```

```{r Get INDEL Summary 2-2 in Test Analysis}
  print(Export_Summary_IndelSV_perFragments(Input = Result_HLA2_INDEL, Mut_IC50_th = 500))
```

```{r Get SV Sample 1 in Test Analysis}
  Result_HLA1_SV <- MainSVFUSIONClass1(input_file = "data/sample_sv_bnd.txt",
                                       file_name_in_hla_table = "sample",
                                       hla_file = "data/sample_hla_table_c1.txt",
                                       refflat_file  = "lib/refFlat.txt",
                                       refmrna_file = "lib/refMrna.fa",
                                       rnaexp_file = "data/sample_rna_exp.txt",
                                       netMHCpan_dir = "lib/netMHCpan-4.0/netMHCpan",
                                       refdna_file = "lib/GRCh37.fa",
                                       mutation_alt_bnd_column = 5,
                                       gene_symbol_column = 7,
                                       mate_id_column = 8)
  print(head(Result_HLA1_SV))
```

```{r Get SV Summary 1-1 in Test Analysis}
  print(Export_Summary_IndelSV(Result_HLA1_SV, Mut_IC50_th = 500))
```

```{r Get SV Summary 1-2 in Test Analysis}
  print(Export_Summary_IndelSV_perFragments(Result_HLA1_SV, Mut_IC50_th = 500))
```

```{r Get SV Sample 2 in Test Analysis}
  Result_HLA2_SV <- MainSVFUSIONClass2(input_file = "data/sample_sv_bnd.txt",
                                       file_name_in_hla_table = "sample",
                                       hla_file = "data/sample_hla_table_c2.txt",
                                       refflat_file  = "lib/refFlat.txt",
                                       refmrna_file = "lib/refMrna.fa",
                                       rnaexp_file = "data/sample_rna_exp.txt",
                                       netMHCIIpan_dir = "lib/netMHCIIpan-3.2/netMHCIIpan",
                                       refdna_file = "lib/GRCh37.fa",
                                       mutation_alt_bnd_column = 5,
                                       gene_symbol_column = 7,
                                       mate_id_column = 8)
  print(head(Result_HLA2_SV))
```

```{r Get SV Summary Sample 2 in Test Analysis}
  print(Export_Summary_IndelSV(Result_HLA2_SV, Mut_IC50_th = 500))
```

```{r Get Fragment Sample 1 in Test Analysis}
  Result_HLA1_Seq <- MainSeqFragmentClass1(input_sequence = "atggcagaagatgatccatatttgggaaggcctgaaaaaatgtttcatttggatccttctttgactcatacaatatttaatc",
                                           file_name_in_hla_table = "sample",
                                           hla_file = "data/sample_hla_table_c1.txt",
                                           hmdir = getwd(),
                                           job_id = "NO_job_id",
                                           refflat_file  = "lib/refFlat.txt",
                                           refmrna_file = "lib/refMrna.fa",
                                           netMHCpan_dir = "lib/netMHCpan-4.0/netMHCpan",
                                           reference_nm_id = c("NM_003998", "NM_001165412"))
  print(head(Result_HLA1_Seq))
```

```{r Get Fragment Summary 1 in Test Analysis}
  print(Export_Summary_Fragments(Result_HLA1_Seq, Mut_IC50_th = 500))
```

```{r Get Fragment Sample 2 in Test Analysis}
  Result_HLA2_Seq <- MainSeqFragmentClass2(input_sequence = "atggcagaagatgatccatatttgggaaggcctgaacaaatgtttcatttgatccttctttgactcatacaatatttaatc",
                                           file_name_in_hla_table = "sample",
                                           hla_file = "data/sample_hla_table_c2.txt",
                                           hmdir = getwd(),
                                           job_id = "NO_job_id",
                                           refflat_file  = "lib/refFlat.txt",
                                           refmrna_file = "lib/refMrna.fa",
                                           netMHCIIpan_dir = "lib/netMHCIIpan-3.2/netMHCIIpan",
                                           reference_gene_symbol = c("NFKB1", "BCL3"))
  print(head(Result_HLA2_Seq))
```
```{r Get Fragment Summary 2 in Test Analysis}
  print(Export_Summary_Fragments(Result_HLA2_Seq, Mut_IC50_th = 500))
```
