<!DOCTYPE html>
<html>
<head>
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<title>SampleCodeForNeoantimon</title>
<style type="text/css">
/**
 * Prism.s theme ported from highlight.js's xcode style
 */
pre code {
  padding: 1em;
}
.token.comment {
  color: #007400;
}
.token.punctuation {
  color: #999;
}
.token.tag,
.token.selector {
  color: #aa0d91;
}
.token.boolean,
.token.number,
.token.constant,
.token.symbol {
  color: #1c00cf;
}
.token.property,
.token.attr-name,
.token.string,
.token.char,
.token.builtin {
  color: #c41a16;
}
.token.inserted {
  background-color: #ccffd8;
}
.token.deleted {
  background-color: #ffebe9;
}
.token.operator,
.token.entity,
.token.url,
.language-css .token.string,
.style .token.string {
  color: #9a6e3a;
}
.token.atrule,
.token.attr-value,
.token.keyword {
  color: #836c28;
}
.token.function,
.token.class-name {
  color: #DD4A68;
}
.token.regex,
.token.important,
.token.variable {
  color: #5c2699;
}
.token.important,
.token.bold {
  font-weight: bold;
}
.token.italic {
  font-style: italic;
}
</style>
<style type="text/css">
body {
  font-family: sans-serif;
  max-width: 800px;
  margin: auto;
  padding: 1em;
  line-height: 1.5;
  box-sizing: border-box;
}
body, .footnotes, code { font-size: .9em; }
li li { font-size: .95em; }
*, *:before, *:after {
  box-sizing: inherit;
}
pre, img { max-width: 100%; }
pre, pre:hover {
  white-space: pre-wrap;
  word-break: break-all;
}
pre code {
  display: block;
  overflow-x: auto;
}
code { font-family: 'DejaVu Sans Mono', 'Droid Sans Mono', 'Lucida Console', Consolas, Monaco, monospace; }
:not(pre) > code, code[class] { background-color: #F8F8F8; }
code.language-undefined, pre > code:not([class]) {
  background-color: inherit;
  border: 1px solid #eee;
}
table {
  margin: auto;
  border-top: 1px solid #666;
}
table thead th { border-bottom: 1px solid #ddd; }
th, td { padding: 5px; }
thead, tfoot, tr:nth-child(even) { background: #eee; }
blockquote {
  color: #666;
  margin: 0;
  padding-left: 1em;
  border-left: 0.5em solid #eee;
}
hr, .footnotes::before { border: 1px dashed #ddd; }
.frontmatter { text-align: center; }
#TOC .numbered li { list-style: none; }
#TOC .numbered { padding-left: 0; }
#TOC .numbered ul { padding-left: 1em; }
table, .body h2 { border-bottom: 1px solid #666; }
.body .appendix, .appendix ~ h2 { border-bottom-style: dashed; }
.footnote-ref a::before { content: "["; }
.footnote-ref a::after { content: "]"; }
section.footnotes::before {
  content: "";
  display: block;
  max-width: 20em;
}

@media print {
  body {
    font-size: 12pt;
    max-width: 100%;
  }
  tr, img { page-break-inside: avoid; }
}
@media only screen and (min-width: 992px) {
  pre { white-space: pre; }
}
</style>
</head>
<body>
<div class="frontmatter">
<div class="title"><h1>SampleCodeForNeoantimon</h1></div>
<div class="author"><h2>T. Hasegawa</h2></div>
<div class="date"><h3>2024/6/12</h3></div>
</div>
<div class="body">
<!--
%\VignetteEngine{knitr::knitr}
%\VignetteIndexEntry{Sample Code to Use Neoantimon}
-->
<h2 id="data-preparation-and-sample-codes-for-analysis">Data Preparation and Sample Codes for Analysis</h2>
<pre><code class="language-r">#install.packages('devtools');
library(devtools);
</code></pre>
<pre><code>## Loading required package: usethis
</code></pre>
<pre><code class="language-r">install_github('hase62/Neoantimon');
</code></pre>
<pre><code>## Skipping install of 'Neoantimon' from a github remote, the SHA1 (68abede1) has not changed since last install.
##   Use `force = TRUE` to force installation
</code></pre>
<pre><code class="language-r">library(Neoantimon);
if (!requireNamespace(&quot;BiocManager&quot;, quietly = TRUE))
    install.packages(&quot;BiocManager&quot;)
BiocManager::install(&quot;biomaRt&quot;)
</code></pre>
<pre><code>## Bioconductor version 3.19 (BiocManager 1.30.23), R 4.4.0 (2024-04-24)
</code></pre>
<pre><code>## Warning: package(s) not installed when version(s) same as or greater than current; use
##   `force = TRUE` to re-install: 'biomaRt'
</code></pre>
<pre><code class="language-r">library(biomaRt)
</code></pre>
<p>To calculate the binding affinity of neoantigen candaites, which are generated from from SNVs, to HLA ClassI. When using MHCflurry, [[1]] and [[2]] include the results of NetMHCpan and MHCflurry, respectively.</p>
<pre><code class="language-r">Result_HLA1_SNV &lt;- MainSNVClass1(input_annovar_format_file = &quot;data/sample_vcf.annovar.txt&quot;,
                                   file_name_in_hla_table = &quot;sample&quot;,
                                   hla_file = &quot;data/sample_hla_table_c1.txt&quot;,
                                   refflat_file  = &quot;lib/refFlat.grch37.txt&quot;,
                                   refmrna_file = &quot;lib/refMrna.grch37.fa&quot;,
                                   rnaexp_file = &quot;data/sample_rna_exp.txt&quot;,
                                   netMHCpan_dir = &quot;lib/netMHCpan-4.1/netMHCpan&quot;,
                                   depth_tumor_column = 12,
                                   depth_normal_column = 14,
                                   SNPs = &quot;data/sample_vcf.snps.txt&quot;,
                                   multiple_variants = TRUE)
</code></pre>
<pre><code>## [1] &quot;Please indicate hla_file and file_name_in_hla_table, or hla_types appropriately.&quot;
</code></pre>
<pre><code class="language-r">  Result_HLA1_SNV &lt;- CalculatePriorityScores(result = Result_HLA1_SNV, useRNAvaf = FALSE)
  print(head(Result_HLA1_SNV))
</code></pre>
<pre><code>##      result P_I P_R P
</code></pre>
<pre><code class="language-r">  print(Export_Summary_SNV(Input = Result_HLA1_SNV, Mut_Rank_th = 5, Wt_Rank_th = 5))
</code></pre>
<pre><code>##              Num_All_Alteration        Num_Evaluated_Alteration 
##                               0                               0 
## Num_Alteration_Generating_NeoAg                 Num_All_Peptide 
##                               0                               0 
##           Num_Evaluated_Peptide    Num_Peptide_Generating_NeoAg 
##                               0                               0
</code></pre>
<p>To calculate the binding affinity of neoantigen candaites, which are generated from from SNVs, to HLA ClassII.</p>
<pre><code class="language-r"> Result_HLA2_SNV &lt;- MainSNVClass2(input_annovar_format_file = &quot;data/sample_vcf.annovar.txt&quot;,
                                   file_name_in_hla_table = &quot;sample&quot;,
                                   hla_file = &quot;data/sample_hla_table_c2.txt&quot;,
                                   refflat_file  = &quot;lib/refFlat.grch37.txt&quot;,
                                   refmrna_file = &quot;lib/refMrna.grch37.fa&quot;,
                                   rnaexp_file = &quot;data/sample_rna_exp.txt&quot;,
                                   netMHCIIpan_dir = &quot;lib/netMHCIIpan-4.3/netMHCIIpan&quot;,
                                   depth_tumor_column = 12,
                                   depth_normal_column = 14,
                                   SNPs = &quot;data/sample_vcf.snps.txt&quot;,
                                   multiple_variants = TRUE)
</code></pre>
<pre><code>## [1] &quot;Please indicate hla_file and file_name_in_hla_table, or hla_types appropriately.&quot;
</code></pre>
<pre><code class="language-r">  Result_HLA2_SNV &lt;- CalculatePriorityScores(result = Result_HLA2_SNV, useRNAvaf = FALSE) 
  print(head(Result_HLA2_SNV))
</code></pre>
<pre><code>##      result P_I P_R P
</code></pre>
<pre><code class="language-r">  print(Export_Summary_SNV(Input = Result_HLA2_SNV, Mut_Rank_th = 10, Wt_Rank_th = 10))
</code></pre>
<pre><code>##              Num_All_Alteration        Num_Evaluated_Alteration 
##                               0                               0 
## Num_Alteration_Generating_NeoAg                 Num_All_Peptide 
##                               0                               0 
##           Num_Evaluated_Peptide    Num_Peptide_Generating_NeoAg 
##                               0                               0
</code></pre>
<p>To calculate the binding affinity of neoantigen candaites, which are generated from from indels, to HLA ClassI. When using MHCflurry, [[1]] and [[2]] include the results of NetMHCpan and MHCflurry, respectively.</p>
<pre><code class="language-r">Result_HLA1_INDEL &lt;- MainINDELClass1(input_annovar_format_file = &quot;data/sample_vcf.annovar.txt&quot;,
                                       file_name_in_hla_table = &quot;sample&quot;,
                                       hla_file = &quot;data/sample_hla_table_c1.txt&quot;,
                                       refflat_file  = &quot;lib/refFlat.grch37.txt&quot;,
                                       refmrna_file = &quot;lib/refMrna.grch37.fa&quot;,
                                       rnaexp_file = &quot;data/sample_rna_exp.txt&quot;,
                                       netMHCpan_dir = &quot;lib/netMHCpan-4.1/netMHCpan&quot;,
                                       depth_tumor_column = 12,
                                       depth_normal_column = 14,
                                       SNPs = &quot;data/sample_vcf.snps.txt&quot;,
                                       multiple_variants = TRUE)
</code></pre>
<pre><code>## [1] &quot;Please indicate hla_file and file_name_in_hla_table, or hla_types appropriately.&quot;
</code></pre>
<pre><code class="language-r">  Result_HLA1_INDEL_1 &lt;- CalculatePriorityScores(result = Result_HLA1_INDEL, useRNAvaf = FALSE)
  print(head(Result_HLA1_INDEL_1))
</code></pre>
<pre><code>##      result P_I P_R P
</code></pre>
<pre><code class="language-r">  print(Export_Summary_IndelSV(Input = Result_HLA1_INDEL_1, Mut_Rank_th = 5))
</code></pre>
<pre><code>##              Num_All_Alteration        Num_Evaluated_Alteration 
##                               0                               0 
## Num_Alteration_Generating_NeoAg                 Num_All_Peptide 
##                               0                               0 
##           Num_Evaluated_Peptide    Num_Peptide_Generating_NeoAg 
##                               0                               0
</code></pre>
<pre><code class="language-r">  print(Export_Summary_IndelSV_perFragments(Input = Result_HLA1_INDEL_1, Mut_Rank_th = 5))
</code></pre>
<pre><code>##                                                    
## Num_Peptide_Per_Pep                                
## Num_Cond_Peptide_Per_Pep                           
## Num_Rest_Peptide_Per_Pep                           
## Num_Rest_Peptide_Per_Pep / Num_Cond_Peptide_Per_Pep
## -logP
</code></pre>
<p>To calculate the binding affinity of neoantigen candaites, which are generated from from indels, to HLA ClassII.</p>
<pre><code class="language-r">  Result_HLA2_INDEL &lt;- MainINDELClass2(input_annovar_format_file = &quot;data/sample_vcf.annovar.txt&quot;,
                                       file_name_in_hla_table = &quot;sample&quot;,
                                       hla_file = &quot;data/sample_hla_table_c2.txt&quot;,
                                       refflat_file  = &quot;lib/refFlat.grch37.txt&quot;,
                                       refmrna_file = &quot;lib/refMrna.grch37.fa&quot;,
                                       rnaexp_file = &quot;data/sample_rna_exp.txt&quot;,
                                       netMHCIIpan_dir = &quot;lib/netMHCIIpan-4.3/netMHCIIpan&quot;,
                                       depth_tumor_column = 12,
                                       depth_normal_column = 14,
                                       SNPs = &quot;data/sample_vcf.snps.txt&quot;,
                                       multiple_variants = TRUE)
</code></pre>
<pre><code>## [1] &quot;Please indicate hla_file and file_name_in_hla_table, or hla_types appropriately.&quot;
</code></pre>
<pre><code class="language-r">  Result_HLA2_INDEL &lt;- CalculatePriorityScores(result = Result_HLA2_INDEL, useRNAvaf = FALSE)  
  print(head(Result_HLA2_INDEL))
</code></pre>
<pre><code>##      result P_I P_R P
</code></pre>
<pre><code class="language-r">  print(Export_Summary_IndelSV(Input = Result_HLA2_INDEL, , Mut_Rank_th = 10))
</code></pre>
<pre><code>##              Num_All_Alteration        Num_Evaluated_Alteration 
##                               0                               0 
## Num_Alteration_Generating_NeoAg                 Num_All_Peptide 
##                               0                               0 
##           Num_Evaluated_Peptide    Num_Peptide_Generating_NeoAg 
##                               0                               0
</code></pre>
<pre><code class="language-r">  print(Export_Summary_IndelSV_perFragments(Input = Result_HLA2_INDEL, , Mut_Rank_th = 5))
</code></pre>
<pre><code>##                                                    
## Num_Peptide_Per_Pep                                
## Num_Cond_Peptide_Per_Pep                           
## Num_Rest_Peptide_Per_Pep                           
## Num_Rest_Peptide_Per_Pep / Num_Cond_Peptide_Per_Pep
## -logP
</code></pre>
<p>To calculate the binding affinity of neoantigen candaites, which are generated from from SVs, to HLA ClassI.</p>
<pre><code class="language-r">  Result_HLA1_SV &lt;- MainSVFUSIONClass1(input_file = &quot;data/sample_sv_bnd.txt&quot;,
                                       file_name_in_hla_table = &quot;sample&quot;,
                                       hla_file = &quot;data/sample_hla_table_c1.txt&quot;,
                                       refflat_file  = &quot;lib/refFlat.grch37.txt&quot;,
                                       refmrna_file = &quot;lib/refMrna.grch37.fa&quot;,
                                       rnaexp_file = &quot;data/sample_rna_exp.txt&quot;,
                                       netMHCpan_dir = &quot;lib/netMHCpan-4.1/netMHCpan&quot;,
                                       refdna_file = &quot;GRCh37.fa&quot;,
                                       mutation_alt_bnd_column = 5,
                                       gene_symbol_column = 7,
                                       mate_id_column = 8)
</code></pre>
<pre><code>## [1] &quot;Please indicate hla_file and file_name_in_hla_table, or hla_types appropriately.&quot;
</code></pre>
<pre><code class="language-r">  Result_HLA1_SV &lt;- CalculatePriorityScores(result = Result_HLA1_SV, useRNAvaf = FALSE)
  print(head(Result_HLA1_SV))
</code></pre>
<pre><code>##      result P_I P_R P
</code></pre>
<pre><code class="language-r">  print(Export_Summary_IndelSV(Result_HLA1_SV, , Mut_Rank_th = 5))
</code></pre>
<pre><code>##              Num_All_Alteration        Num_Evaluated_Alteration 
##                               0                               0 
## Num_Alteration_Generating_NeoAg                 Num_All_Peptide 
##                               0                               0 
##           Num_Evaluated_Peptide    Num_Peptide_Generating_NeoAg 
##                               0                               0
</code></pre>
<pre><code class="language-r">  print(Export_Summary_IndelSV_perFragments(Result_HLA1_SV, , Mut_Rank_th = 5))
</code></pre>
<pre><code>##                                                    
## Num_Peptide_Per_Pep                                
## Num_Cond_Peptide_Per_Pep                           
## Num_Rest_Peptide_Per_Pep                           
## Num_Rest_Peptide_Per_Pep / Num_Cond_Peptide_Per_Pep
## -logP
</code></pre>
<p>To calculate the binding affinity of neoantigen candaites, which are generated from from SVs, to HLA ClassII.</p>
<pre><code class="language-r">  Result_HLA2_SV &lt;- MainSVFUSIONClass2(input_file = &quot;data/sample_sv_bnd.txt&quot;,
                                       file_name_in_hla_table = &quot;sample&quot;,
                                       hla_file = &quot;data/sample_hla_table_c2.txt&quot;,
                                       refflat_file  = &quot;lib/refFlat.grch37.txt&quot;,
                                       refmrna_file = &quot;lib/refMrna.grch37.fa&quot;,
                                       rnaexp_file = &quot;data/sample_rna_exp.txt&quot;,
                                       netMHCIIpan_dir = &quot;lib/netMHCIIpan-4.3/netMHCIIpan&quot;,
                                       refdna_file = &quot;lib/GRCh37.fa&quot;,
                                       mutation_alt_bnd_column = 5,
                                       gene_symbol_column = 7,
                                       mate_id_column = 8)
</code></pre>
<pre><code>## [1] &quot;Please indicate hla_file and file_name_in_hla_table, or hla_types appropriately.&quot;
</code></pre>
<pre><code class="language-r">  Result_HLA2_SV &lt;- CalculatePriorityScores(result = Result_HLA2_SV, useRNAvaf = FALSE)
  print(head(Result_HLA2_SV))
</code></pre>
<pre><code>##      result P_I P_R P
</code></pre>
<pre><code class="language-r">  print(Export_Summary_IndelSV(Result_HLA2_SV, Mut_Rank_th = 5))
</code></pre>
<pre><code>##              Num_All_Alteration        Num_Evaluated_Alteration 
##                               0                               0 
## Num_Alteration_Generating_NeoAg                 Num_All_Peptide 
##                               0                               0 
##           Num_Evaluated_Peptide    Num_Peptide_Generating_NeoAg 
##                               0                               0
</code></pre>
<pre><code class="language-r">  print(Export_Summary_IndelSV_perFragments(Result_HLA2_SV, Mut_Rank_th = 5))
</code></pre>
<pre><code>##                                                    
## Num_Peptide_Per_Pep                                
## Num_Cond_Peptide_Per_Pep                           
## Num_Rest_Peptide_Per_Pep                           
## Num_Rest_Peptide_Per_Pep / Num_Cond_Peptide_Per_Pep
## -logP
</code></pre>
<p>To calculate the binding affinity of neoantigen candaites, which are directly generated from RNA sequences, to HLA ClassI. The peptides included in the original genes (“NM_003998”, “NM_001165412”) are removed from the results.</p>
<pre><code class="language-r">   Result_HLA1_Seq &lt;- MainSeqFragmentClass1(input_sequence = &quot;atggcagaagatgatccatatttgggaaggcctgaaaaaatgtttcatttggatccttctttgactcatacaatatttaatc&quot;,
                                           file_name_in_hla_table = &quot;sample&quot;,
                                           hla_file = &quot;data/sample_hla_table_c1.txt&quot;,
                                           hmdir = getwd(),
                                           job_id = &quot;NO_job_id&quot;,
                                           refflat_file  = &quot;lib/refFlat.grch37.txt&quot;,
                                           refmrna_file = &quot;lib/refMrna.grch37.fa&quot;,
                                           netMHCpan_dir = &quot;lib/netMHCpan-4.1/netMHCpan&quot;,
                                           reference_nm_id = c(&quot;NM_003998&quot;, &quot;NM_001165412&quot;))
</code></pre>
<pre><code>## [1] &quot;Please indicate hla_file and file_name_in_hla_table, or hla_types appropriately.&quot;
</code></pre>
<pre><code class="language-r">  Result_HLA1_Seq &lt;- CalculatePriorityScores(result = Result_HLA1_Seq, useRNAvaf = FALSE)
  print(head(Result_HLA1_Seq))
</code></pre>
<pre><code>##      result P_I P_R P
</code></pre>
<pre><code class="language-r">  #print(Export_Summary_Fragments(Result_HLA1_Seq, Mut_Rank_th = 5))
</code></pre>
<p>To calculate the binding affinity of neoantigen candaites, which are directly generated from RNA sequences, to HLA ClassII. The peptides included in the riginal genes (“NFKB1”, “BCL3”) are removed from the results.</p>
<pre><code class="language-r">  Result_HLA2_Seq &lt;- MainSeqFragmentClass2(input_sequence = &quot;atggcagaagatgatccatatttgggaaggcctgaacaaatgtttcatttgatccttctttgactcatacaatatttaatc&quot;,
                                           file_name_in_hla_table = &quot;sample&quot;,
                                           hla_file = &quot;data/sample_hla_table_c2.txt&quot;,
                                           hmdir = getwd(),
                                           job_id = &quot;NO_job_id&quot;,
                                           refflat_file  = &quot;lib/refFlat.grch37.txt&quot;,
                                           refmrna_file = &quot;lib/refMrna.grch37.fa&quot;,
                                           netMHCIIpan_dir = &quot;lib/netMHCIIpan-4.3/netMHCIIpan&quot;,
                                           reference_gene_symbol = c(&quot;NFKB1&quot;, &quot;BCL3&quot;))
</code></pre>
<pre><code>## [1] &quot;Please indicate hla_file and file_name_in_hla_table, or hla_types appropriately.&quot;
</code></pre>
<pre><code class="language-r">  Result_HLA2_Seq &lt;- CalculatePriorityScores(result = Result_HLA2_Seq, useRNAvaf = FALSE)
  print(head(Result_HLA2_Seq))
</code></pre>
<pre><code>##      result P_I P_R P
</code></pre>
<pre><code class="language-r">  #print(Export_Summary_Fragments(Result_HLA2_Seq, Mut_Rank_th = 5))
</code></pre>
</div>
<script src="https://cdn.jsdelivr.net/npm/prismjs@1.29.0/components/prism-core.min.js" defer></script>
<script src="https://cdn.jsdelivr.net/npm/prismjs@1.29.0/plugins/autoloader/prism-autoloader.min.js" defer></script>
</body>
</html>
